pkg/helm/actions/auth_test.go

```diff
@@ -0,0 +1,141 @@
+package actions
+
+import (
+	"io/ioutil"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/api/helm/v1beta1"
+	"github.com/stretchr/testify/require"
+	"helm.sh/helm/v3/pkg/action"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestAuthenticationDoesNotSetRepoURL(t *testing.T) {
+	tests := []struct {
+		name           string
+		config         *v1beta1.ConnectionConfig
+		hasCA          bool
+		hasTLS         bool
+		expectCaFile   bool
+		expectCertFile bool
+	}{
+		{
+			name: "CA certificate only",
+			config: &v1beta1.ConnectionConfig{
+				URL: "https://charts.example.com",
+				CA:  configv1.ConfigMapNameReference{Name: "test-ca"},
+			},
+			hasCA:        true,
+			expectCaFile: true,
+		},
```

Comment:
You have the `hasCA` case, the `hasTLS` case, and the has-both case...should we have a case for has-neither?